scipy/stats/tests/mpmath_distribution.py

```diff
@@ -0,0 +1,169 @@
+        return u4 / u2**2 - 3
+
+    def _moment(self, order, center=None, **kwargs):
+        def integrand(x):
+            return self._pdf(x, **kwargs)*(x - center)**order
+
+        if center is None:
+            center = self._mean(**kwargs)
+
+        a, b = self._support(**kwargs)
+        return mp.quad(integrand, (a, b))
+
+    def pdf(self, x):
+        fun = np.vectorize(self._pdf)
+        x = self._make_mpf_array(x)
+        res = fun(x, **self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def cdf(self, x):
+        fun = np.vectorize(self._cdf)
+        x = self._make_mpf_array(x)
+        res = fun(x, **self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def sf(self, x):
+        fun = np.vectorize(self._sf)
+        x = self._make_mpf_array(x)
+        res = fun(x, **self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def logpdf(self, x):
+        fun = np.vectorize(self._logpdf)
+        x = self._make_mpf_array(x)
+        res = fun(x, **self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def logcdf(self, x):
+        fun = np.vectorize(self._logcdf)
+        x = self._make_mpf_array(x)
+        res = fun(x, **self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def logsf(self, x):
+        fun = np.vectorize(self._logsf)
+        x = self._make_mpf_array(x)
+        res = fun(x, **self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def support(self):
+        fun = np.vectorize(self._support)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def entropy(self):
+        fun = np.vectorize(self._entropy)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def mean(self):
+        fun = np.vectorize(self._mean)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def var(self):
+        fun = np.vectorize(self._var)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def skew(self):
+        fun = np.vectorize(self._skew)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def kurtosis(self):
+        fun = np.vectorize(self._kurtosis)
+        res = fun(**self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+    def moment(self, order):
+        fun = np.vectorize(self._moment)
+        order = self._make_mpf_array(order)
+        res = fun(order, **self._params)
+        return np.asarray(res, dtype=np.float64)[()]
+
+class Normal(Distribution):
+
+    # always override __init__ so IDEs hint at shape parameter names
+    def __init__(self, *, mu, sigma):
+        super().__init__(mu=mu, sigma=sigma)
+
+    def _pdf(self, x, mu, sigma):
+        return mp.npdf(x, mu, sigma)
+
+class SkewNormal(Distribution):
+
+    def __init__(self, *, a):
```

Comment:
There's really no need to implement `loc` and `scale` parameters if we're using this to test the existing distributions. The reason is that `loc` and `scale` have generic implementations for all distributions, and that is already tested, so if we're interested in the accuracy of a distribution method, we only need to test it for varying *shape* parameters.